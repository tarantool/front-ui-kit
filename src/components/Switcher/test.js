import React from 'react';
import renderer from 'react-test-renderer';
import { Switcher } from './index';

it('Switcher renders correctly', () => {
  const changeState = () => null;

  const tree = renderer.create(
    <>
      <Switcher
        checked
        onChange={changeState}
        title='Initial'
        name='switcherChecked'
      >
        Checked
      </Switcher>
      <Switcher
        onChange={changeState}
        name='switcher'
      >
        Initial
      </Switcher>
      <Switcher
        disabled
        checked
        onChange={changeState}
        name='switcherCheckedDisabled'
      >
        Checked disabled
      </Switcher>
      <Switcher
        disabled
        onChange={changeState}
        name='switcherDisabled'
      >
        Disabled
      </Switcher>
    </>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
