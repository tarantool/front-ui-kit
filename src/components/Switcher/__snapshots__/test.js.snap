// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switcher renders correctly 1`] = `
Array [
  <label
    className="css-s5xdrg"
    title="Initial"
  >
    <input
      checked={true}
      className="css-8j4tix"
      name="switcherChecked"
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="css-6l8wcs"
    />
    <span
      className="css-1ybmbi2"
    >
      Checked
    </span>
  </label>,
  <label
    className="css-s5xdrg"
  >
    <input
      className="css-8j4tix"
      name="switcher"
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="css-znp835"
    />
    <span
      className="css-1ybmbi2"
    >
      Initial
    </span>
  </label>,
  <label
    className="css-s5xdrg"
  >
    <input
      checked={true}
      className="css-8j4tix"
      disabled={true}
      name="switcherCheckedDisabled"
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="css-1tnggdq"
    />
    <span
      className="css-1ybmbi2"
    >
      Checked disabled
    </span>
  </label>,
  <label
    className="css-s5xdrg"
  >
    <input
      className="css-8j4tix"
      disabled={true}
      name="switcherDisabled"
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="css-pc19o9"
    />
    <span
      className="css-1ybmbi2"
    >
      Disabled
    </span>
  </label>,
]
`;
